### ISSUE TITLE
clientValidationEnabled option for child forms

### ISSUE THREAD

**Issue by koichirose** [kristijanhusak/laravel-form-builder#191]
I have a form with children forms, like this:
```php
<?php
class SurveyForm extends Form
{
    public function buildForm()
    {
        $this
            ->add('translations', 'translation', [
                'type' => 'form',
                'wrapper' => [
                    'class' => 'translations'
                ],
                'label' => false,
                'options' => [
                    'class' =>  'App\Forms\SurveyTranslationForm',
                    'label' => false,
                    'wrapper' => false,
                ],
            ]);
            //more fields...
```

Adding `protected $clientValidationEnabled = false;` here works as expected.
I'd like to disable client side validation for the translation child form only, though.

I tried adding the option to my SurveyTranslationForm, but it doesn't seem to work.
Is that possible? How can I do it?

**Comment by kristijanhusak** [comment]
Can you post content of your `translation` form type?

**Comment by koichirose** [comment]
Here it is:
```php
<?php namespace App\Forms;

use Kris\LaravelFormBuilder\Form;

class SurveyTranslationForm extends Form
{
    //protected $clientValidationEnabled = false;

    public function buildForm()
    {
        $this
            ->add('name', 'text', [
                'label' => trans('surveys.name'),
                'rules' => 'required',
            ])
            ->add('id', 'hidden')
            ->add('locale', 'hidden', [
                'rules' => 'required',
            ])
            ;
    }
}
```

**Comment by kristijanhusak** [comment]
@koichirose not the form itself, but the `translation` type. It should be custom field type:

```php
$this
    ->add(
        'translations', 
        'translation', // I need content of this custom field type
        [...]
    );
```

**Comment by koichirose** [comment]
Sorry, here it is:
```
<?php
namespace App\Forms\Fields;

use Kris\LaravelFormBuilder\Fields\CollectionType;

class Translation extends CollectionType {

    protected function getTemplate()
    {
        return 'vendor.laravel-form-builder.translation';
    }

    public function render(array $options = [], $showLabel = true, $showField = true, $showError = true)
    {
        //$options['somedata'] = 'This is some data for view';

        return parent::render($options, $showLabel, $showField, $showError);
    }
}
```

I should move the view out of the 'vendor' folder. The view is the same one I mentioned here: https://github.com/kristijanhusak/laravel-form-builder/issues/190

**Comment by kristijanhusak** [comment]
Thanks. I'll check it out when i catch some time.

**Comment by kristijanhusak** [comment]
Fixed in latest version (1.6.42). You can also pass form option `'client_validation' => false` to child form/collection to disable client validation. This allows disabling it only in some situations, but if you need it disabled all the time, setting property is good solution.

```php
      $this
            ->add('translations', 'translation', [
                'type' => 'form',
                'wrapper' => [
                    'class' => 'translations'
                ],
                'label' => false,
                'options' => [
                    'class' =>  'App\Forms\SurveyTranslationForm',
                    'client_validation' => false, // This
                    'label' => false,
                    'wrapper' => false,
                ],
            ]);
```

Thanks for reporting.

**Comment by koichirose** [comment]
I can confirm it works also by setting the following on the child form:
`protected $clientValidationEnabled = false;`

Thanks!
